### PR TITLE
Hides the symbols except for those that are marked as visible

### PR DESCRIPTION
**Pull request by ematejska**
This fixes the immediate mode issue when the TensoFlow modules was used with swift because the symbols are not being loaded twice for llvm.

https://bugs.swift.org/projects/TF/issues/TF-1273

### CODE REVIEW

**Comment by ematejska** [comment]
This is not to be merged yet because the TFE_GetServerDef symbol in Sources/TensorFlow/Core/Runtime.swift ends up undefined. (It's an experimental symbol in tensorflow/c/c_api_experimental.h whose visibility is not public.

**Comment by compnerd** [comment]
`TFE_GetServerDef` is in TensorFlow and needs to be annotated with default visibility.

File: Sources/CX10/xla_tensor_wrapper.cc [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #if defined(_WIN32)
 #define XLA_API __declspec(dllexport)
 #else
-#define XLA_API
+#define XLA_API __attribute__((visibility("default")))
```
Comment (compnerd) [inline Sources/CX10/xla_tensor_wrapper.cc:18]: Could you please use the reserved spelling?  `__attribute__((__visibility__("default")))`

**Comment by ematejska** [comment]
Yeah, I wonder if TensorFlow will accept that change or if there is anyther API that can be used.

**Comment by ematejska** [comment]
I got the PR for tensorflow here: https://github.com/tensorflow/tensorflow/pull/39551 but if it's accepted, we have to wait for S4TF to pull that version of Tensorflow in before merging this.

**Comment by ematejska** [comment]
Removed reliance of TFE_GetServerDef here: https://github.com/tensorflow/swift-apis/pull/976. If testing passes, we're good to merge.

File: Sources/CX10/xla_tensor_wrapper.cc (hunk 1 above)
Comment (ematejska) [reply Sources/CX10/xla_tensor_wrapper.cc]: Done

**Pull request merged**